src/backends/dlib/dlibinputconns.h

```diff
@@ -0,0 +1,136 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef DLIBINPUTCONNS_H
+#define DLIBINPUTCONNS_H
+
+#include "imginputfileconn.h"
+#include "opencv2/opencv.hpp"
+#include "dlib/data_io.h"
+#include "dlib/image_io.h"
+#include "dlib/image_transforms.h"
+#include "dlib/image_processing.h"
+#include "dlib/opencv/to_open_cv.h"
+#include "dlib/opencv/cv_image.h"
+
+
+#include "inputconnectorstrategy.h"
+#include "ext/base64/base64.h"
+
+namespace dd {
+    class DlibInputInterface {
+    public:
+        DlibInputInterface() {}
+
+        DlibInputInterface(const DlibInputInterface &tii)
+                : _dv(tii._dv), _ids(tii._ids) {}
+
+        ~DlibInputInterface() {}
+
+    public:
+        // parameters common to all Dlib input connectors
+        std::vector<dlib::matrix<dlib::rgb_pixel>> _dv;
+        std::vector<dlib::matrix<dlib::rgb_pixel>> _dv_test;
+        std::vector<std::string> _ids; // input ids (eg. Image Ids).
+    };
+
+    class ImgDlibInputFileConn : public ImgInputFileConn, public DlibInputInterface {
+    public:
+        ImgDlibInputFileConn()
+                : ImgInputFileConn() {
+            reset_dv();
+        }
+
+        ImgDlibInputFileConn(const ImgDlibInputFileConn &i)
+                : ImgInputFileConn(i), DlibInputInterface(i){}
+
+        ~ImgDlibInputFileConn() {}
+
+        int channels() const {
+            if (_bw) return 1;
+            else return 3; // RGB
+        }
+
+        int height() const {
+            return _height;
+        }
+
+        int width() const {
+            return _width;
+        }
+
+        int batch_size() const {
+            if (!_dv.empty())
+                return _dv.size();
+            else return ImgInputFileConn::batch_size();
+        }
+
+        int test_batch_size() const {
+            if (!_dv_test.empty())
+                return _dv_test.size();
+            else return ImgInputFileConn::test_batch_size();
+        }
+
+        void init(const APIData &ad) {
+            ImgInputFileConn::init(ad);
+        }
+
+        void transform(const APIData &ad) {
+            try {
+                ImgInputFileConn::transform(ad);
+            }
+            catch (InputConnectorBadParamException &e) {
+                throw;
+            }
+            for (size_t i = 0; i < _images.size(); i++) {
+                dlib::matrix<dlib::rgb_pixel> inputImg;
+                dlib::assign_image(inputImg, dlib::cv_image<dlib::rgb_pixel>(_images.at(i)));
```

Comment:
Do you believe you should add _bw support ?

Reply:
Good idea, thanks. I'll add that in.

Reply:
In the meanwhile I added support for `_bw` in that if it is specified, it is correctly converted from the `cv::Mat` to a `dlib::matrix` object such that the supported dlib models can accept the image.

Reply:
Yes I believe this is the right behavior for now. Usually the issue is that the image channels need to match the network's input size. Thanks for the code update.

Reply:
Got it 👍 minor tweak to remove unnecessary code but should be good now.